bots/elasticsearch/tasks.py

```diff
@@ -117,8 +117,13 @@ def do_run(self, es_url=None, es_index=None):
     def bulk_stream(self, es_index):
         Source = apps.get_model('share.Source')
         opts = {'_index': es_index, '_type': 'sources'}
-        for source in Source.objects.exclude(long_title='').all():
-            yield {'_op_type': 'index', '_id': source.name, **self.serialize(source), **opts}
+
+        for source in Source.objects.all():
+            # remove sources from search that don't appear on the sources page
+            if source.icon == '':
```

Comment:
No, just `if source.icon:` will do. The field returns a little `FieldFile` instance, not the actual favicon.